garden_ai/_model.py

```diff
@@ -152,10 +162,31 @@ def predict(self, data):
         """
         return self.model.predict(data)
 
+    # Dill serialization breaks using new __getattr__ if missing this.
+    def __getstate__(self):
+        return self.__dict__
+
+    def __getattr__(self, attr):
+        if attr == "_model":
+            # Dill deserialization infinitely recursive without this
+            raise AttributeError()
+        if self._model is not None:
+            if attr in self.__dict__:
+                # Use self definition of attr
+                return self.__getattribute__(attr)
+            else:
+                # self does not have definition of attr
+                # Search _model instead
+                return self._model.__getattribute__(attr)
+        else:
+            # _model is None, lazy load and try again.
+            self._lazy_load_model()
```

Comment:
Are there cases where we could hit this branch before we'd normally need to load the model? Probably not in practice? If so this would be fine. But if there are then I'd make this a little more restrictive so we try 

```
if attr in self.__dict__:
                # Use self definition of attr
                return self.__getattribute__(attr)
```
before loading the model